p2p/protocol/identify/id.go

```diff
@@ -31,9 +33,27 @@ const ID = "/ipfs/id/1.0.0"
 
 // LibP2PVersion holds the current protocol version for a client running this code
 // TODO(jbenet): fix the versioning mess.
+// XXX: Don't change this till 2020. You'll break all go-ipfs versions prior to
+// 0.4.17 which asserted an exact version match.
 const LibP2PVersion = "ipfs/0.1.0"
 
-var ClientVersion = "go-libp2p/3.3.4"
+// ClientVersion is the default user agent.
+//
+// Deprecated: Set this with the UserAgent option.
+var ClientVersion = "github.com/libp2p/go-libp2p"
+
+func init() {
+	bi, ok := debug.ReadBuildInfo()
+	if !ok {
+		return
+	}
+	version := bi.Main.Version
+	if version == "(devel)" {
+		ClientVersion = bi.Main.Path
```

Comment:
let's keep the `devel` designator somewhere in the reported path.

Reply:
Anything built manually (i.e., built from a manually checked-out source tree) will have a `(devel)` version. It doesn't mean it's _actually_ a development version.

To get a proper version, one would have to build with `go get github.com/libp2p/go-libp2p-examples@latest` (i.e., build with go mod).

Users that care can override this.

Reply:
ok.

Reply:
Can we suffix with @unknown?